### PR TITLE
Issue #PS-388 feat: Attendance overview common reusable date-range popup component

### PR DESCRIPTION
**Pull request by rahulg1254**
## Description

Please describe the changes in your pull request in few words here.

### Changes

List the changes done to fix a bug or introducing a new feature.

## How to test

Describe the steps required to test the changes proposed in the pull request.

Please consider using the closing keyword if the pull request is proposed to
fix an issue already created in the repository
(https://help.github.com/articles/closing-issues-using-keywords/)

## Summary by CodeRabbit

- **New Features**
  - Added support for English, Hindi, and Marathi languages in the app.
  - Introduced internationalization (i18n) settings for a more localized experience.

- **Configuration Updates**
  - Updated environment variables for better configuration management.
  - Introduced ESLint and Prettier configurations for code quality and formatting.
  - Updated Node.js version to `16.17.0`.

- **Documentation**
  - Significant updates to the README for improved project setup and configuration instructions.

- **Deployment**
  - Added a GitHub Actions workflow for automated deployment to the development environment.

### CODE REVIEW

File: src/pages/attendance-overview.tsx [hunk 1]
```diff
@@ -54,10 +51,19 @@ const AttendanceOverview: React.FC<AttendanceOverviewProps> = () => {
...
     Array<any>
   >([]);
+  const [selectedValue, setSelectedValue] = React.useState<string>('');
 
   const theme = useTheme<any>();
   const pathname = usePathname();
 
+  const menuItems = [
+    'Last 7 Days(18 - 25 May)',
+    'As Of Today, 25 May',
```
Comment (itsvick) [inline src/pages/attendance-overview.tsx:61]: @rahulg1254 add in translation files

File: src/pages/attendance-overview.tsx [hunk 2]
```diff
@@ -208,197 +214,169 @@ const AttendanceOverview: React.FC<AttendanceOverviewProps> = () => {
...
-            <Typography textAlign={'left'} fontSize={'22px'} m={'1rem'}>
-              {t('Attendance Overview')}
-            </Typography>
-          </Box>
+    <Box>
+      <Header />
+      <Box sx={{ display: 'flex', justifyContent: 'center' }}>
+        <Box width={'100%'}>
+          <Typography textAlign={'left'} fontSize={'22px'} m={'1rem'}>
+            {t('Attendance Overview')}
```
Comment (itsvick) [inline src/pages/attendance-overview.tsx:222]: @rahulg1254 should be in translation file with capital letters

File: src/pages/attendance-overview.tsx [hunk 3]
```diff
@@ -208,197 +214,169 @@ const AttendanceOverview: React.FC<AttendanceOverviewProps> = () => {
...
-            <Grid item xs={7}>
-              <OverviewCard
-                label="Low Attendance Students"
-                value="Bharat Kumar, Ankita Kulkarni, +3 more"
-              />
-            </Grid>
+      <Box display={'flex'} className="card_overview">
+        <Grid container spacing={0}>
+          <Grid item xs={5}>
+            <OverviewCard label="Centre Attendance" value="71%" />
```
Comment (itsvick) [inline src/pages/attendance-overview.tsx:268]: @rahulg1254 label should be translated

File: src/pages/attendance-overview.tsx [hunk 4]
```diff
@@ -208,197 +214,169 @@ const AttendanceOverview: React.FC<AttendanceOverviewProps> = () => {
...
-      </Box>
-    </>
+      {learnerData?.length > 0 ? (
+        <Box>
+          {displayStudentList?.map((user: any) => (
+            <StudentsStatsList
+              key={user.userId}
+              name={user.name}
+              value1={user.present_percentage}
+              value2={user.absent}
```
Comment (itsvick) [inline src/pages/attendance-overview.tsx:361]: @rahulg1254 dont use props as value1 or value2

File: src/components/DateRangePopup.tsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+          <FormControl fullWidth sx={{ m: 1 }}>
+            <Select
+              sx={{ height: '32px' }}
+              value={selectedValue}
+              displayEmpty
+              onClick={handleModalOpen}
+              inputProps={{ readOnly: true }}
+            >
+              <MenuItem value="" disabled>
+                Select an option
```
Comment (itsvick) [inline src/components/DateRangePopup.tsx:70]: @rahulg1254 translation

File: src/components/DateRangePopup.tsx [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+              </MenuItem>
+              <MenuItem value={selectedValue}>
+                {selectedValue ? selectedValue : 'Select an option'}
```
Comment (itsvick) [inline src/components/DateRangePopup.tsx:73]: @rahulg1254 translation

File: src/components/DateRangePopup.tsx [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+          sx={modalStyle}
+          gap="10px"
+          display="flex"
+          flexDirection="column"
+          borderRadius={'1rem'}
+        >
+          <Box>
+            <Grid container>
+              <Grid item xs={6}>
+                <Typography textAlign={'left'}>Date Range</Typography>
```
Comment (itsvick) [inline src/components/DateRangePopup.tsx:96]: @rahulg1254 translation

File: src/components/DateRangePopup.tsx [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+                    <Check fontSize="small" />
+                  </ListItemIcon>
+                )}
+                {item}
+              </MenuItem>
+            ))}
+          </MenuList>
+          <Divider sx={dividerStyle} />
+          <Button variant="contained" onClick={handleModalClose}>
+            Apply
```
Comment (itsvick) [inline src/components/DateRangePopup.tsx:133]: @rahulg1254 translation

File: src/pages/attendance-overview.tsx (hunk 4 above)
Comment (rahulg1254) [reply src/pages/attendance-overview.tsx]: resolved

File: src/components/DateRangePopup.tsx (hunk 6 above)
Comment (rahulg1254) [reply src/components/DateRangePopup.tsx]: resolve

File: src/components/DateRangePopup.tsx (hunk 8 above)
Comment (rahulg1254) [reply src/components/DateRangePopup.tsx]: resolve

File: src/components/DateRangePopup.tsx (hunk 5 above)
Comment (rahulg1254) [reply src/components/DateRangePopup.tsx]: resolve

**Pull request merged**